### DISCUSSION TITLE
How to import all Tailwind colors without getting a warning?

### DISCUSSION THREAD

**Discussion by iaseth** [tailwindlabs/tailwindcss#11127] [Help] [answered]
I created a npm package [html-color-names](https://github.com/iaseth/html-color-names) which provides color palettes based on each of the standard 148 HTML colors. I want the user to be able to import 1-2 colors from my package along with all Tailwind colors like this:

```js
const tailwindcolors = require('tailwindcss/colors');

const hcn = require("html-color-names");
const { htmlcolors } = hcn;
const { firebrick } = htmlcolors; // destructure all the HTML colors you want to use

export default {
	content: [],
	theme: {
		colors: {
			firebrick, // HTML color
			...tailwindcolors, // Tailwind colors
		},
		extend: {},
	},
	plugins: [],
}
```

This works but shows these warnings on compilation:
```
warn - As of Tailwind CSS v2.2, `lightBlue` has been renamed to `sky`.
warn - Update your configuration file to silence this warning.

warn - As of Tailwind CSS v3.0, `warmGray` has been renamed to `stone`.
warn - Update your configuration file to silence this warning.

warn - As of Tailwind CSS v3.0, `trueGray` has been renamed to `neutral`.
warn - Update your configuration file to silence this warning.

warn - As of Tailwind CSS v3.0, `coolGray` has been renamed to `gray`.
warn - Update your configuration file to silence this warning.

warn - As of Tailwind CSS v3.0, `blueGray` has been renamed to `slate`.
warn - Update your configuration file to silence this warning.
```
I am guessing this is because `tailwindcss` contains some colors which have been renamed in later versions. Is there a way to import all 22 current Tailwind colors at once without getting a warning?

**Comment by wongjn** [comment, marked as answer]
The non-deprecated colors are included in default configuration, so you could look at using `theme.extend.colors` instead like:

```diff
 content: [],
 theme: {
-	colors: {
-		firebrick, // HTML color
-		...tailwindcolors, // Tailwind colors
-	extend: {},
+	extend: {
+		colors: {
+			firebrick,
+		},
	},
	plugins: [],
}

**Reply by iaseth** [reply]
It worked! Thanks for responding so fast!

**Reply by ivan-kleshnin** [reply]
@wongjn I used `tailwindcss/colors` **in browser** like below:

```tsx
import colors from "tailwindcss/colors"

...

export function Circle(props: CircleProps) {
  const {color, size} = props
  return <div className="rounded-full"
    style={{
      width: `${size}rem`,
      height: `${size}rem`,
      // @ts-ignore
      backgroundColor: `${colors[color[0]][color[1]]}`
    }}
  />
}
```

Now this warning. The alt. approach proposed in docs https://tailwindcss.com/docs/configuration#referencing-in-java-script is notably bad

> Note that this will transitively pull in a lot of our build-time dependencies, resulting in bigger client-side bundle size. To avoid this, we recommend using a tool like [babel-plugin-preval](https://github.com/kentcdodds/babel-plugin-preval) to generate a static version of your configuration at build-time.

And another mentioned alternative of https://github.com/kentcdodds/babel-plugin-preval is not maintaned as Babel is less relevant nowaydas.

A plugin to expose vars in CSS https://github.com/omarkhatibco/tailwind-css-variables does not work with modern tooling as well. 

So the remaining way is to manually repeat names and values in `index.css`?

---

I understand that for TW values it's better to use `className` getting class modifier support and whatnot. 
But for nonterministic colors and sizes it's still necessary to fallback to direct low-level styling.

**Comment by mitchdav** [comment]
We faced a similar issue, where we want an object with all of the colors that we can spread over.

This logic will give you an object with just the non-deprecated colors, without triggering the deprecation warnings.

```js
const colors = require("tailwindcss/colors");

// Tailwind's default color palette includes colors that have been renamed, and if you spread over these, it will
// trigger warnings. This code extracts the non-getter properties from the object, so that we don't trigger warnings.

const entries = Object.entries(Object.getOwnPropertyDescriptors(colors)).filter(
  ([key, descriptor]) => {
    return typeof descriptor.get !== "function"; // Exclude getters
  }
);

const nonDeprecatedColors = Object.fromEntries(
  entries.map(([key, descriptor]) => [key, descriptor.value])
);
```
